lib/Doctrine/DBAL/Driver/Mysqli/MysqliStatement.php

```diff
@@ -228,6 +245,63 @@ public function execute($params = null)
         return true;
     }
 
+    /**
+     * Move all stream resources that are bound with the LARGE_OBJECT type
+     * from the _bindesValues array over to longData so that we can
+     * send them using send_long_data.
+     */
+    private function prepareLongData()
+    {
+        foreach ($this->_bindedValues as $parameter => $value) {
+            if ($this->types[$parameter - 1] !== static::$_paramTypeMap[ParameterType::LARGE_OBJECT]) {
+                continue;
+            }
+
+            if (is_resource($value)) {
+                if (get_resource_type($value) !== 'stream') {
+                    throw new InvalidArgumentException('Resources passed with the LARGE_OBJECT parameter type must be stream resources.');
+                }
+                $this->_bindedValues[$parameter] = null;
```

Comment:
I don't think it's a good idea to change the object state at the statement execution time. If somebody decides to reuse an existing stream and an existing statement, it won't work.

Instead of changing the object state, this method could return an array like `[$boundValues, $longData]`, each of which would be processed accordingly. This way, the state of the object will remain intact, and the `$longData` property will be no longer needed.